go/vt/vttablet/tabletmanager/shard_sync.go

```diff
@@ -154,18 +156,24 @@ func (tm *TabletManager) shardSyncLoop(ctx context.Context, notifyChan <-chan st
 // If the shard record indicates a new master has taken over, this returns
 // success (we successfully synchronized), but the returned masterAlias will be
 // different from the input tablet.Alias.
-func syncShardMaster(ctx context.Context, ts *topo.Server, tablet *topodatapb.Tablet, masterTermStartTime time.Time) (masterAlias *topodatapb.TabletAlias, err error) {
+func syncShardMaster(ctx context.Context, ts *topo.Server, tablet *topodatapb.Tablet, masterTermStartTime time.Time) (masterAlias *topodatapb.TabletAlias, haveOutdatedInfo bool, err error) {
 	ctx, cancel := context.WithTimeout(ctx, *topo.RemoteOperationTimeout)
 	defer cancel()
 
 	var shardInfo *topo.ShardInfo
+	haveOutdatedInfo = false
```

Comment:
This confused me at first. Can you rename this to `shouldDemote`?
It is not that the tablet has "outdated" info. It is that the tablet's earlier timestamp means that it is no longer the primary.

Reply:
changed